### ISSUE TITLE
[New rule] Prefer defining type for generic types

### ISSUE THREAD

**Issue by NickBalnaves** [dart-code-checker/dart-code-metrics#771]
**Please describe what the rule should do:**
To prefer using a type when a generic type is specified in a class. 
In the Flutter source a `Future` is define as `abstract class Future<T>` and when creating a function it is possible to write `Future newFunction() {}`, but it should be enforced to define the type with the generic type defined such as `Future<void>`. 

**What category of rule is this? (place an "X" next to just one item)**
[ ] Warns about a potential error (problem)
[X] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about (it will be better if you can provide both good and bad examples):**

Bad:
```dart
Future newDialog() =>
    showDialog<bool>(
      context: context,
      builder: (context) => AlertDialog(
        title: Text('Test'),
      ),
    );
```

Good:
```dart
Future<bool?> newDialog() =>
    showDialog<bool>(
      context: context,
      builder: (context) => AlertDialog(
        title: Text('Test'),
      ),
    );
```

**Are you willing to submit a pull request to implement this rule?**
No

**Comment by incendial** [comment]
@NickBalnaves have you tried https://github.com/dart-lang/language/blob/master/resources/type-system/strict-raw-types.md? This should help

**Comment by NickBalnaves** [comment]
@incendial Exactly what I want! Thanks for the link
